plugins/module_utils/module_helper.py

```diff
@@ -151,23 +220,32 @@ def __init__(self, module=None):
         if module:
             self.module = module
 
-        if not isinstance(module, AnsibleModule):
+        if isinstance(self.module, dict):
+            if self.named_deprecations:
+                self.module['argument_spec'].update(self.make_ack_named_deprecation_arg_spec())
             self.module = AnsibleModule(**self.module)
 
+        for name, deprecation in self.named_deprecations.items():
+            deprecation.prepare(name, self.module, self.ack_deprecations_module_param)
+
+        for ack in self.module.params.get(self.ack_deprecations_module_param) or []:
+            self.ack_named_deprecation(ack)
+
     def update_output(self, **kwargs):
-        if kwargs:
-            self.output_dict.update(kwargs)
+        self.output_dict.update(kwargs or {})
```

Comment:
If `kwargs` is falsey here, it will be `{}` (or equivalent to it). So this can just be
```suggestion
        self.output_dict.update(kwargs)
```
(Same for the one below.)

Reply:
Thanks for spotting that one up. Coming up in the next push.